src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -1172,18 +1528,38 @@ def find_existing_acr(cmd, app: "ContainerApp"):
     return None, None
 
 
-def check_env_name_on_rg(cmd, managed_env, resource_group_name, location):
+def check_env_name_on_rg(cmd, env, resource_group_name, location, custom_location_id=None, connected_cluster_id=None):
+    env_name = env
+    resource_type = None
+    if is_valid_resource_id(env):
+        env_dict = parse_resource_id(env)
+        env_name = env_dict.get("name")
+        resource_group_name = env_dict.get("resource_group")
+        resource_type = env_dict.get("resource_type")
+    if resource_type is None:
+        if custom_location_id or connected_cluster_id:
+            resource_type = CONNECTED_ENVIRONMENT_TYPE
     if location:
         _ensure_location_allowed(cmd, location, CONTAINER_APPS_RP, "managedEnvironments")
```

Comment:
>_ensure_location_allowed [](http://example.com/codeflow?start=8&length=24)

should we check location according to resource type?

Reply:
Done.